src/common/state/redis.c

```diff
@@ -380,32 +578,32 @@ void task_log_redis_callback(redisAsyncContext *c, void *r, void *privdata) {
     return;
   }
   /* Otherwise, parse the task and call the callback. */
-  task_log_subscribe_data *data = callback_data->data;
+  task_table_subscribe_data *data = callback_data->data;
 
-  task_instance *instance = malloc(reply->element[2]->len);
-  memcpy(instance, reply->element[2]->str, reply->element[2]->len);
+  task *task = malloc(payload->len);
+  memcpy(task, payload->str, payload->len);
   if (data->subscribe_callback) {
-    data->subscribe_callback(instance, data->subscribe_context);
+    data->subscribe_callback(task, data->subscribe_context);
   }
-  task_instance_free(instance);
+  free_task(task);
 }
 
-void redis_task_log_subscribe(table_callback_data *callback_data) {
+void redis_task_table_subscribe(table_callback_data *callback_data) {
   db_handle *db = callback_data->db_handle;
-  task_log_subscribe_data *data = callback_data->data;
+  task_table_subscribe_data *data = callback_data->data;
 
-  if (memcmp(&data->node.id[0], &NIL_ID.id[0], UNIQUE_ID_SIZE) == 0) {
-    redisAsyncCommand(db->sub_context, task_log_redis_callback,
-                      (void *) callback_data->timer_id,
-                      "PSUBSCRIBE task_log:*:%d", data->state_filter);
+  if (IS_NIL_ID(data->node)) {
```

Comment:
What I mean is that the functionality required from the "subscribe to everything" case might be different than what we have here.